### PR TITLE
Support for TCRseq and template update

### PR DESCRIPTION
**Pull request by ggabernet**
- Template update to v12.1
- Add support for 5'RACE data
- Add support for TCRseq data

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/bcellmagic branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/bcellmagic)
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Documentation in `docs` is updated
- [x] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/bcellmagic/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Review by FriederikeHanssen — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 * [#69](https://github.com/nf-core/bcellmagic/pull/69): Template update to nf-core tools v1.10.2
 * [#69](https://github.com/nf-core/bcellmagic/pull/69): Added parameter json schema
 * [#74](https://github.com/nf-core/bcellmagic/pull/74): Added possibility of setting UMI start position and better UMI docs
+* Added primer handling params: `--vprimer_start`, `--cprimer_start`, `--primer_mask_mode`, `--primer_maxeror`, `--primer_consensus`
+* Template update to nf-core tools v1.12.1
+* Added support for TCR data with params: `--loci`
+* Added support for mice data and possibility for other species with params: `--species`
+* Added support for 5' RACE technologiy with params: `--race_5prime`, `--race_linker`
```
Comment (apeltzer) [inline CHANGELOG.md:17]:
```suggestion
* Added support for 5' RACE technology with params: `--race_5prime`, `--race_linker`
```

File: README.md [hunk 2]
```diff
@@ -38,6 +38,15 @@ The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool
 
 See [usage docs](https://nf-co.re/bcellmagic/usage) for all of the available options when running the pipeline.
 
+## Pipeline Summary
```
Comment (apeltzer) [inline README.md:41]: This probably needs an update :)

File: README.md [hunk 3]
```diff
@@ -38,6 +38,15 @@ The pipeline is built using [Nextflow](https://www.nextflow.io), a workflow tool
...
+
+By default, the pipeline currently performs the following:
+
+<!-- TODO nf-core: Fill in short bullet-pointed list of default steps of pipeline -->
+
+* Sequencing quality control (`FastQC`)
+* Overall pipeline run summaries (`MultiQC`)
+
```
Comment (apeltzer) [inline README.md:49]: If I remember well, this does much more :D

File: README.md [hunk 4]
```diff
@@ -46,13 +55,18 @@ The nf-core/bcellmagic pipeline comes with documentation about the pipeline: [us
 
 nf-core/bcellmagic was originally written by Gisela Gabernet, Simon Heumos, Alexander Peltzer.
 
+We thank the following people for their extensive assistance in the development
+of this pipeline:
+
+<!-- TODO nf-core: If applicable, make list of people who have also contributed -->
```
Comment (apeltzer) [inline README.md:61]: Include acknowledgements to all people who were / are involved :-)

File: README.md [hunk 5]
```diff
@@ -65,3 +79,7 @@ You can cite the `nf-core` publication as follows:
 >
 > _Nat Biotechnol._ 2020 Feb 13. doi: [10.1038/s41587-020-0439-x](https://dx.doi.org/10.1038/s41587-020-0439-x).
 > ReadCube: [Full Access Link](https://rdcu.be/b1GjZ)
+
+In addition, references of tools and data used in this pipeline are as follows:
+
+<!-- TODO nf-core: Add bibliography of tools and data used in your pipeline -->
```
Comment (apeltzer) [inline README.md:85]: Immcantation , ....

**Review by apeltzer — APPROVED** [review]
Minor points, should be fine afterwards :-)

File: README.md (hunk 3 above)
Comment (ggabernet) [reply README.md:49]: ah ok sorry, I missed the summary in the README, must be new from one of the latest template updates!

File: README.md (hunk 4 above)
Comment (ggabernet) [reply README.md:61]: Here is still mostly the two of us and Simon, we are already mentioned as main pipeline developers above.

File: README.md (hunk 2 above)
Comment (ggabernet) [reply README.md:41]: yes right we moved it from Output to Readme

File: README.md (hunk 5 above)
Comment (ggabernet) [reply README.md]: added the references now!

**Comment by ggabernet** [comment]
@apeltzer thanks for spotting the missing things in the README, fixed now!

**Comment by apeltzer** [comment]
No problem, thanks for putting in so much work 🎉

**Pull request merged**